mitodl/odl-video-service
# add YouTubeVideo model admin features

#### What are the relevant tickets?
https://github.com/mitodl/odl-video-service/issues/500

#### What's this PR do?
1. adds status filtering
2. adds searching by youtube id, video key
3. adds creation_date to model for sorting

#### How should this be manually tested?
1. Run migrations
2. Whip up YouTubeVideo instances. Here's some shell food:
```
import ui.models
import ui.factories
import random
from django.contrib.auth.models import User

u = User.objects.all()[0]
collections = [ui.factories.CollectionFactory(owner=u) for i in range(2)]
for c in collections:
    c.save()

statuses = ['uploaded', 'uploading']

for i in range(10):
    v = ui.factories.VideoFactory(collection=collections[i % 2])
    v.save()
    yt = ui.models.YouTubeVideo(video=v, id="yt_%s" % i, status=statuses[i % 2])
    yt.save()

```
3. Go to http://localhost:8089/admin/ui/youtubevideo/
4. Verify that you can filter by status, search by youtube_id and video_key, sort by creation_date
